### DISCUSSION TITLE
How to make Volar unwrap refs in <template>?

### DISCUSSION THREAD

**Discussion by fabis94** [vuejs/language-tools#1186] [Q&A]
I'm working on a Nuxt 2 (w/ bridge) project and it's my understanding that [Vue unwraps refs ](https://vuejs.org/api/composition-api-setup.html#basic-usage) when they're accessed through `this` or in the template. Volar, however, doesn't understand this and reports that the type of `Ref` values in the template is still a `Ref`, not the actual unwrapped value.

<img width="368" alt="image" src="https://user-images.githubusercontent.com/938316/163219131-4813200a-84d8-49c6-b032-f56ebb796d28.png">

How do I resolve this?

**Comment by fabis94** [comment]
Additionally, it doesn't look like refs defined in `<script setup>` are available in the other `<script>` block at all. Is this expected behaviour?

<img width="787" alt="image" src="https://user-images.githubusercontent.com/938316/163220811-935502ec-f02e-428c-b652-4c2fe9b66941.png">

**Comment by GamerGirlandCo** [comment]
seconded. not gonna mince words here, this is fucking annoying

**Comment by matthew-dean** [comment]
I just ran into this with Nuxt 3 / Volar when using `useFetch`. I intuited that the problem was that what was returned as `Ref` within Nuxt was _**not from the same version of Vue as what Volar detected as Vue.**_

Here's what I did to fix it:
1. Upgrade `nuxt`
2. I added `"resolutions": { "vue": "~3.4.15" }` in my `package.json` so that all packages were using the same version of Vue. (This only works if you're using PNPM (best) or Yarn (meh) as a package manager, AFAIK.)

Problem was solved.
